numpy/distutils/conv_template.py

```diff
@@ -216,9 +216,9 @@ def replace(match):
         name = match.group(1)
         try :
             val = env[name]
-        except KeyError:
+        except KeyError as e:
             msg = 'line %d: no definition of key "%s"'%(line, name)
-            raise ValueError(msg)
+            raise ValueError(msg) from e
```

Comment:
Please reread the instructions in #15986 for deciding whether to use `from e` or `from None`.